src/typing/macroContext.ml

```diff
@@ -664,29 +663,21 @@ let type_macro ctx mode cpath f (el:Ast.expr list) p =
 		let index = ref (-1) in
 		let constants = List.map (fun e ->
 			let p = snd e in
-			let e = (try
-				let e' = Texpr.type_constant_value ctx.com.basic e in
-				let rec loop e = match e with
-					| { eexpr = TConst (TString _); epos = p } when Lexer.is_fmt_string p ->
-						Lexer.remove_fmt_string p;
-						todo := (fun() -> Lexer.add_fmt_string p) :: !todo;
-					| _ -> Type.iter loop e
-				in
-				loop e';
-				e
-			with Error (Custom _,_) ->
-				(* if it's not a constant, let's make something that is typed as haxe.macro.Expr - for nice error reporting *)
-				(EBlock [
-					(EVars [("__tmp",null_pos),false,Some (CTPath ctexpr,p),Some (EConst (Ident "null"),p)],p);
-					(EConst (Ident "__tmp"),p);
-				],p)
-			) in
+			let e =
+				if Texpr.is_constant_value ctx.com.basic e then
+					match e with
+					(* temporarily disable format strings processing for macro call argument typing since we want to pass raw constants *)
+					| (EConst (String (s,SSingleQuotes)),p) -> (EConst (String (s,SDoubleQuotes)), p)
```

Comment:
I think the problem with this is that `isFormatExpr` on such macro arguments would return false.

Reply:
Nope, because this expr is only used for argument type unifications. The actual values are encoded later.

Reply:
Right... then I don't remember what the problem here was.

Reply:
Welp, let's see what CI has to say.

Reply:
Hm, actuall I think this breaks #7688 again and we need to recurse there...

Reply:
Also reminder to myself: add test for #1284 while I'm at it.